omc3/check_corrections.py

```diff
@@ -0,0 +1,612 @@
+                         nargs="+",
+                         choices=OPTICS_PARAMS_CHOICES,
+                         help=f"List of parameters for which the cuts should be applied (e.g. {BETA}X {BETA}Y)", )
+    params.add_parameter(name="modelcut",
+                         nargs="+",
+                         type=float,
+                         help="Reject BPMs whose deviation to the model is higher "
+                              "than the corresponding input. Input in order of optics_params.",)
+    params.add_parameter(name="errorcut",
+                         nargs="+",
+                         type=float,
+                         help="Reject BPMs whose error bar is higher than the corresponding "
+                              "input. Input in order of optics_params.",)
+    params.add_parameter(name="beta_filename",
+                         default=CORRECTION_DEFAULTS["beta_filename"],
+                         help="Prefix of the beta file to use. E.g.: ``beta_phase_``", )
+    # Plotting -----------------------------------------------------------------
+    params.add_parameter(name="plot",
+                         action="store_true",
+                         help="Activate plotting."
+                         )
+    params.update(get_plotting_style_parameters())
+    return params
+
+
+@entrypoint(get_params())
+def correction_test_entrypoint(opt: DotDict, accel_opt) -> None:
+    """ Entrypoint function to test the given corrections.
+
+    .. todo:: Instead of writing everything out, it could return
+             dictionaries of the TFSDataFrames and Figures.
+             But I don't see a usecase at the moment (jdilly 2023)
+    """
+    LOG.info("Starting Correction Test.")
+    save_config(Path(opt.output_dir), opt, __file__, unknown_opt=accel_opt)
+
+    opt = _check_opt_add_dicts(opt)
+    opt.output_dir.mkdir(parents=True, exist_ok=True)
+
+    measurement = OpticsMeasurement(opt.meas_dir)
+
+    # get model and filter data
+    accel_inst = manager.get_accelerator(accel_opt)
+    accel_inst.model = _maybe_add_coupling_to_model(accel_inst.model, measurement)
+
+    meas_masks = _get_measurement_filter(
+        accel_inst.model,
+        opt.get_subdict(["meas_dir", "optics_params", "modelcut", "errorcut", "beta_filename"])
+    )
+
+    # sort corrections input (e.g. by folders)
+    corrections = _get_corrections(opt.corrections, opt.file_pattern)
+
+    # run through all scenarios
+    for correction_name, correction_files in corrections.items():
+        _create_model_and_write_diff_to_measurements(
+            opt.output_dir,
+            measurement,
+            correction_name,
+            correction_files,
+            accel_inst,
+            meas_masks,
+        )
+
+    if opt.plot:
+        _do_plots(corrections, opt)
+
+
+# Input Parameters -------------------------------------------------------------
+
+def _check_opt_add_dicts(opt: DotDict) -> DotDict:  # acts inplace...
```

Comment:
Do a `copy.deepcopy` first to not act inplace? The function returns anyway, which really suggests it doesn't act inplace.

Reply:
I don't trust deepcopies. But I did as you asked